presto-main/src/main/java/com/facebook/presto/execution/scheduler/TopologyAwareNodeSelector.java

```diff
@@ -113,7 +113,7 @@ public InternalNode selectCurrentNode()
     }
 
     @Override
-    public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTask> existingTasks)
+    public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTask> existingTasks, int limit)
```

Comment:
Why doesn't the `TopologyAwareNodeSelector` respect the limit?

Reply:
I am not sure if it makes sense to use limit here. The topology awere selector selects the best nodes for splits for maximize locality. I am not sure we can even apply a limit here.

Reply:
@cemcayiroglu : Note even for `SimpleNodeSelector`, when the splits doesn't support remote access (e..g Raptor), the limit is not applied there. cc @highker 

On the other hand, for `TopologyAwareNodeSelector`, the current algorithm makes best effort to co-locate data and compute nodes. Thus we should still be able to apply a limit over the candidate nodes and do best-effort topology-aware selection. Does that make sense?

Reply:
Agree, let's hide `limit` from those non-applicable cases.